### PR TITLE
More tests and docstrings improvements for the config setter class

### PR DESCRIPTION
**Pull request by mchurichi**
Signed-off-by: Maximiliano Churichi <maximiliano.churichi@hpe.com>

### CODE REVIEW

File: test/test_config.py [hunk 1]
```diff
@@ -30,7 +30,15 @@ def test_read_socket_from_environment_variables():
     os.environ['SPIFFE_ENDPOINT_SOCKET'] = fake_socket
 
     setter = ConfigSetter()
-    del os.environ['SPIFFE_ENDPOINT_SOCKET']
```
Comment (scotte) [inline test/test_config.py:33]: Does the environment variable still get deleted at the end of testing? Just want to make sure environment state is reset at the end is all. Thanks!

**Review by scotte — APPROVED** [review]
I have just one question about making sure the env is clean at the end, otherwise looks good to me.

File: test/test_config.py (hunk 1 above)
Comment (mchurichi) [reply test/test_config.py:33]: yes, there's a post step for this suite that restores the environment state after each test run: https://github.com/HewlettPackard/py-spiffe/pull/33/files#diff-45fa5f8530eb815c0cf606032e587adedb04dc70e59b203535d51c44654cfc4dR7

Comment (mchurichi) [reply test/test_config.py:33]: now that you mention this, I think this is a nice util function that we can reuse across test suites

Comment (scotte) [reply test/test_config.py:33]: Ah thanks, that makes sense why I didn't see it now. Thanks!

**Review by glaucimar — APPROVED** [review]
Looks good to me.

File: src/pyspiffe/config.py [hunk 2]
```diff
@@ -8,17 +8,23 @@
 
 
 class Config:
-    """
-    Represents the configuration for a Workload API client
+    """Represents the configuration for a Workload API client
```
Comment (Andres-GC) [inline src/pyspiffe/config.py:11]: nit: _The summary line should end with period, question or exclamation mark._ Both Summary lines in this file do not have any of those ending characters.

Comment (mchurichi) [reply src/pyspiffe/config.py:11]: good catch!

**Review by Andres-GC — APPROVED** [review]
:medal_military:

**Review by glaucimar — APPROVED** [review]

**Review by scotte — APPROVED** [review]
Re-approve

**Pull request merged**